### PR TITLE
[cartoonnetwork] Fix for the recent conversion from CVP to NGTV

### PR DESCRIPTION
**Pull request by shaneallgeier**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Cartoon Network (and most Turner companies) seem to have moved away from CVP in favor of NGTV. This PR implements the new NGTV method in the CN extractor. I wasn't able to find any pages where NGTV didn't work, so I felt comfortable dropping support for the older CVP method. Let me know if you have any questions/concerns and thanks for the review.

### CODE REVIEW

File: youtube_dl/extractor/cartoonnetwork.py [hunk 1]
```diff
@@ -25,18 +23,24 @@ class CartoonNetworkIE(TurnerBaseIE):
...
-        id_type, video_id = re.search(r"_cnglobal\.cvp(Video|Title)Id\s*=\s*'([^']+)';", webpage).groups()
-        query = ('id' if id_type == 'Video' else 'titleId') + '=' + video_id
-        return self._extract_cvp_info(
-            'http://www.cartoonnetwork.com/video-seo-svc/episodeservices/getCvpPlaylist?networkName=CN2&' + query, video_id, {
-                'secure': {
-                    'media_src': 'http://androidhls-secure.cdn.turner.com/toon/big',
-                    'tokenizer_src': 'https://token.vgtf.net/token/token_mobile',
-                },
-            }, {
+        video_id = self._search_regex(r"_cnglobal\.(cvpVideo|cvpTitle|videoMedia)Id\s*=\s*'([^']+)';", webpage, 'video_id', group=2)
```
Comment (dstftw) [inline youtube_dl/extractor/cartoonnetwork.py]: Don't capture groups you don't use.

File: youtube_dl/extractor/cartoonnetwork.py [hunk 2]
```diff
@@ -25,18 +23,24 @@ class CartoonNetworkIE(TurnerBaseIE):
...
+        title = self._html_search_regex(r'og:title.*\scontent="([^"]+)"', webpage, 'title', group=1)
```
Comment (dstftw) [inline youtube_dl/extractor/cartoonnetwork.py]: `_og_search_*`.

File: youtube_dl/extractor/cartoonnetwork.py [hunk 3]
```diff
@@ -25,18 +23,24 @@ class CartoonNetworkIE(TurnerBaseIE):
...
+        description = self._html_search_regex(r'og:description.*\scontent="([^"]+)"', webpage, 'description', default=None, group=1)
```
Comment (dstftw) [inline youtube_dl/extractor/cartoonnetwork.py]: `_og_search_*`.

**Comment by JamesTheAwesomeDude** [comment]
Running c17af258168bc75dc283265c675f286256ff7a0c:

````
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-v', '--hls-prefer-native', 'https://www.cartoonnetwork.com/video/unlocked/when-it-rains-episode.html', '--write-pages']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2018.08.04
[debug] Python version 3.5.2 (CPython) - Linux-4.10.0-38-generic-x86_64-with-Ubuntu-16.04-xenial
[debug] exe versions: ffmpeg 3.4.1-1, ffprobe 3.4.1-1
[debug] Proxy map: {}
[CartoonNetwork] when-it-rains: Downloading webpage
[CartoonNetwork] Saving request to when-it-rains_https_-_www.cartoonnetwork.com_video_unlocked_when-it-rains-episode.html.dump
[CartoonNetwork] 0e9df1345ed47078df3cf75bec73f74562a4fdb8: Downloading JSON metadata
[CartoonNetwork] Saving request to 0e9df1345ed47078df3cf75bec73f74562a4fdb8_http_-_medium.ngtv.io_media_0e9df1345ed47078df3cf75bec73f74562a4fdb8_tv.dump
ERROR: This video is only available for users of participating TV providers. Use --ap-mso to specify Adobe Pass Multiple-system operator Identifier and --ap-username and --ap-password or --netrc to provide account credentials.
Traceback (most recent call last):
  File "/home/james/.local/lib/python3.5/site-packages/youtube_dl/YoutubeDL.py", line 792, in extract_info
    ie_result = ie.extract(url)
  File "/home/james/.local/lib/python3.5/site-packages/youtube_dl/extractor/common.py", line 502, in extract
    ie_result = self._real_extract(url)
  File "/home/james/.local/lib/python3.5/site-packages/youtube_dl/extractor/cartoonnetwork.py", line 46, in _real_extract
    'auth_required': auth_required,
  File "/home/james/.local/lib/python3.5/site-packages/youtube_dl/extractor/turner.py", line 212, in _extract_ngtv_info
    m3u8_url, media_id, ap_data or {}, tokenizer_query)
  File "/home/james/.local/lib/python3.5/site-packages/youtube_dl/extractor/turner.py", line 40, in _add_akamai_spe_token
    query['accessToken'] = self._extract_mvpd_auth(ap_data['url'], content_id, ap_data['site_name'], ap_data['site_name'])
  File "/home/james/.local/lib/python3.5/site-packages/youtube_dl/extractor/adobepass.py", line 1409, in _extract_mvpd_auth
    raise_mvpd_required()
  File "/home/james/.local/lib/python3.5/site-packages/youtube_dl/extractor/adobepass.py", line 1374, in raise_mvpd_required
    'and --ap-username and --ap-password or --netrc to provide account credentials.', expected=True)
youtube_dl.utils.ExtractorError: This video is only available for users of participating TV providers. Use --ap-mso to specify Adobe Pass Multiple-system operator Identifier and --ap-username and --ap-password or --netrc to provide account credentials.

````

* [`0e9df1345ed47078df3cf75bec73f74562a4fdb8_http_-_medium.ngtv.io_media_0e9df1345ed47078df3cf75bec73f74562a4fdb8_tv.dump`](http://ix.io/1m08)
* [`when-it-rains_https_-_www.cartoonnetwork.com_video_unlocked_when-it-rains-episode.html.dump`](http://ix.io/1m9b)

**Comment by JamesTheAwesomeDude** [comment]
Just a smol FYI: this fork does **not** (yet) work:

````
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-v', '--hls-prefer-native', 'https://www.cartoonnetwork.com/video/unlocked/cat-fingers-episode.html']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2018.07.04
[debug] Python version 3.5.2 (CPython) - Linux-4.10.0-38-generic-x86_64-with-Ubuntu-16.04-xenial
[debug] exe versions: ffmpeg 3.4.1-1, ffprobe 3.4.1-1
[debug] Proxy map: {}
[CartoonNetwork] cat-fingers: Downloading webpage
ERROR: Unable to extract video_id; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/home/james/.local/lib/python3.5/site-packages/youtube_dl/YoutubeDL.py", line 792, in extract_info
    ie_result = ie.extract(url)
  File "/home/james/.local/lib/python3.5/site-packages/youtube_dl/extractor/common.py", line 501, in extract
    ie_result = self._real_extract(url)
  File "/home/james/.local/lib/python3.5/site-packages/youtube_dl/extractor/cartoonnetwork.py", line 26, in _real_extract
    video_id = self._search_regex(r"_cnglobal\.(cvpVideo|cvpTitle|videoMedia)Id\s*=\s*'([^']+)';", webpage, 'video_id', group=2)
  File "/home/james/.local/lib/python3.5/site-packages/youtube_dl/extractor/common.py", line 971, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
youtube_dl.utils.RegexNotFoundError: Unable to extract video_id; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
````

BUT, #17271, which has been labeled as "duplicate", **is** now functional.

So unless there's a compelling reason to pull @shaneallgeier's fork over @TheDaChicken's (maybe this guy's code is cleaner? Only 1 cleanly-labeled commit instead of fifteen vague ones is definitely a point in this one's favor), might it be worth reassigning this one as the duplicate and the other as primary/active/candidate?

**Pull request closed without merging**